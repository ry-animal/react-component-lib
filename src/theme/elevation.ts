export const ELEVATION_LIGHT = {
  low: "0px 2px 8px RGBA(0, 0, 0, 0.04)",
  medium: "0px 2px 16px RGBA(0, 0, 0, 0.16)",
  high: "0px 2px 24px RGBA(0, 0, 0, 0.24)",
};

export const ELEVATION_DARK = {
  low: "inherit",
  medium: "inherit",
  high: "inherit",
};
