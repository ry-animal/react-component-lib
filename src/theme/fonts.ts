import "@fontsource/open-sans/400.css";
import "@fontsource/open-sans/500.css";
import "@fontsource/open-sans/600.css";
import "@fontsource/open-sans/700.css";
import "@fontsource/poppins/400.css";
import "@fontsource/poppins/500.css";
import "@fontsource/poppins/600.css";
import "@fontsource/poppins/700.css";

export const FontSize = {
  XXXXL: "72px",
  XXXL: "56px",
  XXL: "40px",
  XL: "32px",
  LG: "24px",
  MD: "20px",
  SM: "18px",
  XS: "16px",
  XXS: "14px",
  XXXS: "12px",
};

/*
 * https://developer.mozilla.org/en-US/docs/Web/CSS/font-weight
 * 400 is regular, 500 is medium, 700 is bold
 */
export const FontWeight = {
  NORMAL: 400,
  MEDIUM: 500,
  SEMI_BOLD: 600,
  BOLD: 700,
};

export const FONT = {
  family: {
    title: "'Poppins', sans-serif",
    body: "'Poppins', sans-serif",
  },
  size: {
    ...FontSize,
  },
  weight: {
    ...FontWeight,
  },
};

export const FontFamily = {
  SANS_SERIF_TITLE: FONT.family.title,
  SANS_SERIF_BODY: FONT.family.body,
};
