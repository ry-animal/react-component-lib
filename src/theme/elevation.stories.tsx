import styled from "styled-components";
import { Text } from "../components/Text";
import { ELEVATION_LIGHT } from "./elevation";

const BoxContainer = styled.div`
  display: flex;
  padding: ${({ theme }) => theme.space.s24};
  flex-direction: row;
  margin-bottom: ${({ theme }) => theme.space.s8};
`;

interface BoxProps {
  item: string;
}
const Box = styled.div<BoxProps>`
  align-self: center;
  background-color: ${({ theme }) => theme.color.SURFACE_2};
  margin-right: ${({ theme }) => theme.space.s32};
  width: 300px;
  height: 300px;
  box-shadow: ${({ theme, item }) => {
    if (item === "low") {
      return theme.elevation.low;
    }
    if (item === "high") {
      return theme.elevation.high;
    }
    return theme.elevation.medium;
  }};
`;

export default {
  title: "Theme/Elevation",
  parameters: { options: { showPanel: false } },
};

const ElevationItem = ({ name, value }: { name: string; value: string }) => (
  <BoxContainer>
    <Box item={name} />
    <div>
      <Text>
        <b>Key:</b> {name}
      </Text>
      <Text>
        <b>Value:</b> {value}
      </Text>
    </div>
  </BoxContainer>
);

export const ElevationList = (): JSX.Element[] =>
  Object.entries(ELEVATION_LIGHT).map(([key, value]) => (
    <ElevationItem key={key} name={key} value={value} />
  ));
