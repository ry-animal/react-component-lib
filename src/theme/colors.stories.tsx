/* eslint-disable no-magic-numbers */
import * as React from "react";
import { withDesign } from "storybook-addon-designs";
import { Switch, Text } from "../";
import { PALETTE_DARK, PALETTE_LIGHT } from "./colors";

export default {
  title: "Theme/Colors",
  decorators: [withDesign],
};

const colorBlock = {
  marginBottom: 32,
} as React.CSSProperties;

const Item = ({
  itemKey,
  itemValue,
  color,
}: {
  itemKey: string;
  itemValue: string;
  color: string;
}) => {
  const renderStyles = {
    margin: 0,
    padding: "32px 16px 16px",
    backgroundColor: itemValue,
  } as React.CSSProperties;

  return (
    <div>
      <div style={renderStyles}>
        <Text text={itemKey} color={color} weight="700" reset />
        <Text
          text={itemValue}
          variant="Caption"
          style={{
            color,
            textTransform: "uppercase",
          }}
          reset
        />
      </div>
    </div>
  );
};

const getKey = (position: number) => Object.keys(PALETTE_LIGHT)[position];
const getColors = (isDark: boolean) => (isDark ? PALETTE_DARK : PALETTE_LIGHT);

interface ColorGroup {
  isDark?: boolean;
}

function Primary({ isDark = false }: ColorGroup) {
  return (
    <div style={colorBlock}>
      <Text
        text="Primary"
        variant="Display4"
        color={getColors(isDark).TEXT_1}
      />
      <Item
        itemKey={getKey(0)}
        itemValue={getColors(isDark).PRIMARY_1}
        color={isDark ? "black" : "white"}
      />
      <Item
        itemKey={getKey(1)}
        itemValue={getColors(isDark).PRIMARY_2}
        color={isDark ? "black" : "white"}
      />
      <Item
        itemKey={getKey(2)}
        itemValue={getColors(isDark).PRIMARY_3}
        color={isDark ? "white" : "black"}
      />
    </div>
  );
}

function Secondary({ isDark = false }: ColorGroup) {
  return (
    <div style={colorBlock}>
      <Text
        text="Secondary"
        variant="Display4"
        color={getColors(isDark).TEXT_1}
      />
      <Item
        itemKey={getKey(3)}
        itemValue={getColors(isDark).SECONDARY_1}
        color="black"
      />
      <Item
        itemKey={getKey(4)}
        itemValue={getColors(isDark).SECONDARY_2}
        color="black"
      />
      <Item
        itemKey={getKey(5)}
        itemValue={getColors(isDark).SECONDARY_3}
        color="black"
      />
    </div>
  );
}

function Success({ isDark = false }: ColorGroup) {
  return (
    <div style={colorBlock}>
      <Text
        text="Success"
        variant="Display4"
        color={getColors(isDark).TEXT_1}
      />
      <Item
        itemKey={getKey(6)}
        itemValue={getColors(isDark).SUCCESS_1}
        color={isDark ? "white" : "black"}
      />
      <Item
        itemKey={getKey(7)}
        itemValue={getColors(isDark).SUCCESS_2}
        color="white"
      />
      <Item
        itemKey={getKey(8)}
        itemValue={getColors(isDark).SUCCESS_3}
        color="white"
      />
    </div>
  );
}

function Error({ isDark = false }: ColorGroup) {
  return (
    <div style={colorBlock}>
      <Text text="Error" variant="Display4" color={getColors(isDark).TEXT_1} />
      <Item
        itemKey={getKey(9)}
        itemValue={getColors(isDark).ERROR_1}
        color={isDark ? "white" : "black"}
      />
      <Item
        itemKey={getKey(10)}
        itemValue={getColors(isDark).ERROR_2}
        color="white"
      />
      <Item
        itemKey={getKey(11)}
        itemValue={getColors(isDark).ERROR_3}
        color="white"
      />
    </div>
  );
}

function Surface({ isDark = false }: ColorGroup) {
  return (
    <div style={colorBlock}>
      <Text
        text="Surface"
        variant="Display4"
        color={getColors(isDark).TEXT_1}
      />
      <Item
        itemKey={getKey(12)}
        itemValue={getColors(isDark).SURFACE_1}
        color={isDark ? "white" : "black"}
      />
      <Item
        itemKey={getKey(13)}
        itemValue={getColors(isDark).SURFACE_2}
        color={isDark ? "white" : "black"}
      />
      <Item
        itemKey={getKey(14)}
        itemValue={getColors(isDark).SURFACE_3}
        color={isDark ? "white" : "black"}
      />
      <Item
        itemKey={getKey(15)}
        itemValue={getColors(isDark).SURFACE_4}
        color={isDark ? "white" : "black"}
      />
    </div>
  );
}

function Texto({ isDark = false }: ColorGroup) {
  return (
    <div style={colorBlock}>
      <Text text="Text" variant="Display4" color={getColors(isDark).TEXT_1} />
      <Item
        itemKey={getKey(16)}
        itemValue={getColors(isDark).TEXT_1}
        color={isDark ? "black" : "white"}
      />
      <Item
        itemKey={getKey(17)}
        itemValue={getColors(isDark).TEXT_2}
        color={isDark ? "black" : "white"}
      />
      <Item
        itemKey={getKey(18)}
        itemValue={getColors(isDark).TEXT_3}
        color={isDark ? "black" : "white"}
      />
      <Item
        itemKey={getKey(19)}
        itemValue={getColors(isDark).TEXT_4}
        color={isDark ? "black" : "white"}
      />
    </div>
  );
}

function Basic({ isDark = false }: ColorGroup) {
  return (
    <div style={colorBlock}>
      <Text text="Basic" variant="Display4" color={getColors(isDark).TEXT_1} />
      <Item
        itemKey={getKey(20)}
        itemValue={getColors(isDark).WHITE}
        color="black"
      />
      <Item
        itemKey={getKey(21)}
        itemValue={getColors(isDark).BLACK}
        color="white"
      />
    </div>
  );
}

export const LightThemePalette = () => (
  <div
    style={{
      backgroundColor: PALETTE_LIGHT.SURFACE_1,
      margin: -16,
      padding: 16,
    }}
  >
    <Primary />
    <Secondary />
    <Success />
    <Error />
    <Surface />
    <Texto />
    <Basic />
  </div>
);
LightThemePalette.parameters = {
  design: {
    type: "figma",
    url: "https://www.figma.com/file/YzgT8FhyEaHw5s1rimOnkn/NFT-Marketplace-v1?node-id=10847%3A6766",
  },
};

export const DarkThemePalette = () => (
  <div
    style={{
      backgroundColor: PALETTE_DARK.SURFACE_1,
      margin: -16,
      padding: 16,
    }}
  >
    <Primary isDark />
    <Secondary isDark />
    <Success isDark />
    <Error isDark />
    <Surface isDark />
    <Texto isDark />
    <Basic isDark />
  </div>
);

DarkThemePalette.parameters = {
  design: {
    type: "figma",
    url: "https://www.figma.com/file/YzgT8FhyEaHw5s1rimOnkn/NFT-Marketplace-v1?node-id=10847%3A6825",
  },
};

export const ThemeSwitcher = () => {
  const [isDark, setIsdark] = React.useState(false);
  const handleRadioChange = () => setIsdark(!isDark);

  return (
    <>
      <div style={colorBlock}>
        <Switch
          checked={isDark}
          label="Dark Mode"
          onChange={handleRadioChange}
        />
      </div>
      {isDark ? <DarkThemePalette /> : <LightThemePalette />}
    </>
  );
};
ThemeSwitcher.parameters = {
  design: {
    type: "figma",
    url: "https://www.figma.com/file/YzgT8FhyEaHw5s1rimOnkn/NFT-Marketplace-v1?node-id=10158%3A130155",
  },
};
