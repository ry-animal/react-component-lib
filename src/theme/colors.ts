import { ThemeColors, ThemeName } from "./theme.interface";

export const lightPaletteColors = {
  PRIMARY_1: "#000000",
  PRIMARY_2: "#5F5F5F",
  PRIMARY_3: "#EEEEEE",

  SECONDARY_1: "#FFF6DA",
  SECONDARY_2: "#FFE8A1",
  SECONDARY_3: "#FFD968",

  SUCCESS_1: "#F2FBF8",
  SUCCESS_2: "#00B176",
  SUCCESS_3: "#007D53",

  ERROR_1: "#FFF2F4",
  ERROR_2: "#BC2828",
  ERROR_3: "#710601",
};

export const lightPaletteBase = {
  SURFACE_1: "#F9F9F9",
  SURFACE_2: "#FFFFFF",
  SURFACE_3: "#EEEEEE",
  SURFACE_4: "#DBDBDB",

  TEXT_1: "#151515",
  TEXT_2: "#404040",
  TEXT_3: "#5F5F5F",
  TEXT_4: "#A4A4A4",
};

export const darkPaletteColors = {
  PRIMARY_1: "#F9F9F9",
  PRIMARY_2: "#EEEEEE",
  PRIMARY_3: "#151515",

  SECONDARY_1: "#FFE8A1",
  SECONDARY_2: "#FFD968",
  SECONDARY_3: "#B38402",

  SUCCESS_1: "#19C68C",
  SUCCESS_2: "#00B176",
  SUCCESS_3: "#007D53",

  ERROR_1: "#FFF2F4",
  ERROR_2: "#BC2828",
  ERROR_3: "#710601",
};

export const darkPaletteBase = {
  SURFACE_1: "#151515",
  SURFACE_2: "#212322",
  SURFACE_3: "#404040",
  SURFACE_4: "#5F5F5F",

  TEXT_1: "#FFFFFF",
  TEXT_2: "#EEEEEE",
  TEXT_3: "#DBDBDB",
  TEXT_4: "#C3C3C3",
};

export const palletteBase = {
  BLACK: "#000000",
  WHITE: "#ffffff",
};

export const PALETTE_LIGHT: ThemeColors = {
  ...lightPaletteColors,
  ...lightPaletteBase,
  ...palletteBase,
};

export const PALETTE_DARK: ThemeColors = {
  ...darkPaletteColors,
  ...darkPaletteBase,
  ...palletteBase,
};

export function getColorsByTheme(theme: ThemeName): ThemeColors {
  if (theme === ThemeName.DARK) {
    return PALETTE_DARK;
  }

  return PALETTE_LIGHT;
}
