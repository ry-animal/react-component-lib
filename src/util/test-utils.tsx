import { render, RenderOptions } from "@testing-library/react";
import "jest-styled-components";
import * as React from "react";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import { LIGHT_THEME } from "../theme";

interface ProvidersProps {
  children: React.ReactNode;
}

function Providers({ children }: ProvidersProps) {
  return (
    <BrowserRouter>
      <ThemeProvider theme={LIGHT_THEME}>{children}</ThemeProvider>
    </BrowserRouter>
  );
}

const customRender = (
  ui: JSX.Element,
  options?:
    | RenderOptions<
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement
      >
    | undefined
) => render(ui, { wrapper: Providers, ...options });

// Duplicated instead of import "../../__mocks__/matchMediaMock"; because jest has trouble resolving imports
Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

// Re-export everything
export * from "@testing-library/react";
// Override render method
export { customRender as render };
