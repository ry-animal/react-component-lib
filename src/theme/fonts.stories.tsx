import * as React from "react";
import { lightPaletteColors } from "./colors";
import { FONT } from "./fonts";

export default {
  title: "Theme/Fonts",
  parameters: { options: { showPanel: false } },
};

const codeStyles = {
  backgroundColor: lightPaletteColors.PRIMARY_3,
  padding: "2px 8px",
  marginBottom: "62px",
  fontSize: "14px",
  fontFamily: "Lucida Console, Courier, monospace",
} as React.CSSProperties;

const Item = ({
  itemKey,
  itemValue,
  fontType,
}: {
  itemKey: string;
  itemValue: string;
  fontType: string;
}) => {
  const renderStyles = {
    margin: 0,
    lineHeight: 1,
    fontFamily: fontType === "family" ? itemValue : FONT.family.title,
    fontSize: fontType === "size" ? itemValue : FONT.size.XXL,
    fontWeight: fontType === "weight" ? itemValue : FONT.weight.SEMI_BOLD,
  } as React.CSSProperties;

  return (
    <>
      <h3 style={renderStyles}>The quick brown fox jumps over the lazy dog</h3>
      <pre style={codeStyles}>
        Key: {itemKey} - Value: {itemValue}
      </pre>
    </>
  );
};

const mapFonts = (obj: object, fontType: string): React.ReactNode => {
  const fonts: React.ReactNodeArray = [];
  Object.entries(obj).forEach(([key, value]) =>
    fonts.push(
      <Item itemKey={key} itemValue={value} fontType={fontType} key={key} />
    )
  );

  return fonts;
};

export const Family = () => mapFonts(FONT.family, "family");

export const Size = () => mapFonts(FONT.size, "size");

export const Weight = () => mapFonts(FONT.weight, "weight");
