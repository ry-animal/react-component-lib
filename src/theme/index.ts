import { getColorsByTheme } from "./colors";
import { ELEVATION_DARK, ELEVATION_LIGHT } from "./elevation";
import { FONT, FontFamily, FontSize, FontWeight } from "./fonts";
import { GRADIENT } from "./gradient";
import { LAYERS } from "./layers";
import { MEDIA_WIDTHS } from "./media";
import { OPACITY } from "./opacity";
import { RADIUS } from "./radius";
import { SPACE } from "./space";
import { ThemeName } from "./theme.interface";

const sharedThemeKeys = {
  radius: {
    ...RADIUS,
  },
  space: {
    ...SPACE,
  },
  type: {
    ...FONT,
  },
  layer: {
    ...LAYERS,
  },
  media: {
    ...MEDIA_WIDTHS,
  },
  opacity: {
    ...OPACITY,
  },
  gradient: {
    ...GRADIENT,
  },
};

export const lightColors = getColorsByTheme(ThemeName.LIGHT);
export const darkColors = getColorsByTheme(ThemeName.DARK);

export const LIGHT_THEME = {
  name: ThemeName.LIGHT,
  ...sharedThemeKeys,
  color: {
    ...lightColors,
  },
  elevation: {
    ...ELEVATION_LIGHT,
  },
};

export const DARK_THEME = {
  name: ThemeName.DARK,
  ...sharedThemeKeys,
  color: {
    ...darkColors,
  },
  elevation: {
    ...ELEVATION_DARK,
  },
};

// Legacy
export { FontFamily, FontSize, FontWeight, MEDIA_WIDTHS, SPACE };
