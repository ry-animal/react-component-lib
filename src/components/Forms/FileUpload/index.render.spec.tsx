import { FileUpload } from ".";
import HangryCat from "../../../assets/images/hangrycat.jpg";
import { lightColors } from "../../../theme";
import { fireEvent, render, screen } from "../../../util/test-utils";

describe("<FileUpload />", () => {
  global.URL.createObjectURL = jest.fn(
    () => "blob:http://localhost:6006/af2e50bd-bc94-4dff-ae8c-440eabdac5c6"
  );
  const sampleProps = () => ({
    allowedMimeTypes: [
      "image/png",
      "image/jpeg",
      "text/plain",
      "model/gltf-binary",
    ],
    maxFileSizeInMB: 100,
    onChange: jest.fn(),
    onError: jest.fn(),
  });

  describe("rendering", () => {
    test("Default rendering matches snapshot", () => {
      expect(
        render(<FileUpload {...sampleProps()} />).asFragment()
      ).toMatchSnapshot();
    });

    test("Initial state rendering", () => {
      expect(
        render(
          <FileUpload
            {...sampleProps()}
            defaultValue={{
              name: "HangryCat",
              mimeType: "image/jpg",
              mediaUri: HangryCat,
            }}
          />
        ).asFragment()
      ).toMatchSnapshot();
    });

    test("Disable drag and drop", () => {
      expect(
        render(
          <FileUpload {...sampleProps()} disableDragAndDrop />
        ).asFragment()
      ).toMatchSnapshot();
    });

    test("Error rendering", async () => {
      const error = "test error";
      expect(
        render(<FileUpload {...sampleProps()} error={error} />).asFragment()
      ).toMatchSnapshot();

      const errorText = await screen.findByRole("alert");
      const dropTarget = await screen.findByRole("application");

      expect(errorText).toHaveStyleRule("color", lightColors.ERROR_2);
      expect(dropTarget).toHaveStyleRule(
        "border",
        `2px dashed ${lightColors.ERROR_2}`
      );
    });

    test("Custom content rendering", async () => {
      const label = "Drop zone label";
      expect(
        render(
          <FileUpload {...sampleProps()} label={label}>
            <p>I am custom content</p>
          </FileUpload>
        ).asFragment()
      ).toMatchSnapshot();

      const dropTarget = await screen.findByRole("application");
      expect(fireEvent.dragEnter(dropTarget)).toBeFalsy();

      expect(dropTarget).toHaveStyleRule("content", `"${label}"`, {
        modifier: "&:after",
      });

      fireEvent.dragLeave(dropTarget);

      expect(dropTarget).not.toHaveStyleRule("content", `"${label}"`, {
        modifier: "&:after",
      });
    });

    test("renders circle dropzone", async () => {
      const { rerender } = render(
        <FileUpload {...sampleProps()}>
          <p>I am custom content</p>
        </FileUpload>
      );
      const dropTarget = await screen.findByRole("application");
      fireEvent.dragEnter(dropTarget);

      expect(dropTarget).not.toHaveStyleRule("border-radius", "10000px", {
        modifier: "&:after",
      });

      rerender(
        <FileUpload {...sampleProps()} isCircleDropzone>
          <p>I am custom content</p>
        </FileUpload>
      );

      expect(dropTarget).toHaveStyleRule("border-radius", "10000px", {
        modifier: "&:after",
      });
    });

    test("Hover state rendering", async () => {
      render(<FileUpload {...sampleProps()} />);
      const dropTarget = await screen.findByRole("application");
      expect(fireEvent.dragEnter(dropTarget)).toBeFalsy();
      expect(dropTarget).toHaveStyleRule(
        "background-color",
        lightColors.SURFACE_2
      );
      fireEvent.dragLeave(dropTarget);
      expect(dropTarget).toHaveStyleRule(
        "background-color",
        lightColors.SURFACE_1
      );
    });
  });
});
