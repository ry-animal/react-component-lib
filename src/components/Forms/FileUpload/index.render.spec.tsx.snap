// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileUpload /> rendering Custom content rendering 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  display: inline-block;
  width: 100%;
  color: #151515;
}

.c1 {
  display: none;
  width: 0;
}

<div
    class="c0"
    label="Drop zone label"
    role="application"
  >
    <p>
      I am custom content
    </p>
    <input
      accept="image/png,image/jpeg,text/plain,model/gltf-binary,.glb,.gltf"
      aria-invalid="false"
      aria-label="File selector"
      class="c1"
      type="file"
    />
  </div>
</DocumentFragment>
`;

exports[`<FileUpload /> rendering Default rendering matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #F9F9F9;
  border: 1px dashed #DBDBDB;
  border-radius: 4px;
  font-family: 'Poppins',sans-serif;
  padding: 3em;
  text-align: center;
}

.c5 {
  display: none;
  width: 0;
}

.c1 {
  color: #404040;
  margin: 1em 30% 1em 30%;
}

.c4 {
  color: #A4A4A4;
  font-weight: 700;
  font-size: 0.8em;
  margin: 1em;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex: unset;
  -ms-flex: unset;
  flex: unset;
  line-height: 1;
  gap: 8px;
}

.c2 {
  text-align: center;
  border: none;
  outline: none;
  overflow: hidden;
  cursor: pointer;
  pointer-events: auto;
  -webkit-transition: background-color 0.25s ease-out;
  transition: background-color 0.25s ease-out;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  min-height: 40px;
  font-weight: 600;
  padding: 0 28px;
  font-size: 16px;
  border-radius: 4px;
  background-color: transparent;
  box-shadow: inset 0 0 0 1px #000000;
  color: #000000;
}

.c2:disabled {
  opacity: 20%;
  cursor: default;
  pointer-events: none;
}

.c2:hover:not([disabled]),
.c2:focus-visible:not([disabled]) {
  box-shadow: inset 0 0 0 2px #000000;
}

<div
    class="c0"
    role="application"
  >
    <div
      class="c1"
    >
      Drag and drop your media files here or click the “Select Media” button below
    </div>
    <button
      aria-label="Select Media"
      class="c2"
    >
      <span
        class="c3"
      >
        Select Media
      </span>
    </button>
    <div
      class="c4"
    >
      Max file size: 100MB
    </div>
    <input
      accept="image/png,image/jpeg,text/plain,model/gltf-binary,.glb,.gltf"
      aria-invalid="false"
      aria-label="File selector"
      class="c5"
      type="file"
    />
  </div>
</DocumentFragment>
`;

exports[`<FileUpload /> rendering Disable drag and drop 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #F9F9F9;
  border: 1px dashed #DBDBDB;
  border-radius: 4px;
  font-family: 'Poppins',sans-serif;
  padding: 3em;
  text-align: center;
}

.c5 {
  display: none;
  width: 0;
}

.c1 {
  color: #404040;
  margin: 1em 30% 1em 30%;
}

.c4 {
  color: #A4A4A4;
  font-weight: 700;
  font-size: 0.8em;
  margin: 1em;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex: unset;
  -ms-flex: unset;
  flex: unset;
  line-height: 1;
  gap: 8px;
}

.c2 {
  text-align: center;
  border: none;
  outline: none;
  overflow: hidden;
  cursor: pointer;
  pointer-events: auto;
  -webkit-transition: background-color 0.25s ease-out;
  transition: background-color 0.25s ease-out;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  min-height: 40px;
  font-weight: 600;
  padding: 0 28px;
  font-size: 16px;
  border-radius: 4px;
  background-color: transparent;
  box-shadow: inset 0 0 0 1px #000000;
  color: #000000;
}

.c2:disabled {
  opacity: 20%;
  cursor: default;
  pointer-events: none;
}

.c2:hover:not([disabled]),
.c2:focus-visible:not([disabled]) {
  box-shadow: inset 0 0 0 2px #000000;
}

<div
    class="c0"
    role="application"
  >
    <div
      class="c1"
    >
      Drag and drop your media files here or click the “Select Media” button below
    </div>
    <button
      aria-label="Select Media"
      class="c2"
    >
      <span
        class="c3"
      >
        Select Media
      </span>
    </button>
    <div
      class="c4"
    >
      Max file size: 100MB
    </div>
    <input
      accept="image/png,image/jpeg,text/plain,model/gltf-binary,.glb,.gltf"
      aria-invalid="false"
      aria-label="File selector"
      class="c5"
      type="file"
    />
  </div>
</DocumentFragment>
`;

exports[`<FileUpload /> rendering Error rendering 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #F9F9F9;
  border: 2px dashed #BC2828;
  border-radius: 4px;
  font-family: 'Poppins',sans-serif;
  padding: 3em;
  text-align: center;
}

.c5 {
  display: none;
  width: 0;
}

.c1 {
  color: #404040;
  margin: 1em 30% 1em 30%;
}

.c4 {
  color: #A4A4A4;
  font-weight: 700;
  font-size: 0.8em;
  margin: 1em;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex: unset;
  -ms-flex: unset;
  flex: unset;
  line-height: 1;
  gap: 8px;
}

.c2 {
  text-align: center;
  border: none;
  outline: none;
  overflow: hidden;
  cursor: pointer;
  pointer-events: auto;
  -webkit-transition: background-color 0.25s ease-out;
  transition: background-color 0.25s ease-out;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  min-height: 40px;
  font-weight: 600;
  padding: 0 28px;
  font-size: 16px;
  border-radius: 4px;
  background-color: transparent;
  box-shadow: inset 0 0 0 1px #000000;
  color: #000000;
}

.c2:disabled {
  opacity: 20%;
  cursor: default;
  pointer-events: none;
}

.c2:hover:not([disabled]),
.c2:focus-visible:not([disabled]) {
  box-shadow: inset 0 0 0 2px #000000;
}

<div
    class="c0"
    role="application"
  >
    <div
      class="c1"
    >
      Drag and drop your media files here or click the “Select Media” button below
    </div>
    <button
      aria-label="Select Media"
      class="c2"
    >
      <span
        class="c3"
      >
        Select Media
      </span>
    </button>
    <div
      class="c4"
    >
      Max file size: 100MB
    </div>
    <input
      accept="image/png,image/jpeg,text/plain,model/gltf-binary,.glb,.gltf"
      aria-invalid="true"
      aria-label="File selector"
      class="c5"
      type="file"
    />
  </div>
  .c0 {
  color: #BC2828;
  font-weight: 500;
  font-size: 12px;
  line-height: 22px;
}

<span
    class="c0"
    role="alert"
  >
    test error
  </span>
</DocumentFragment>
`;

exports[`<FileUpload /> rendering Initial state rendering 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #F9F9F9;
  border: 1px dashed #DBDBDB;
  border-radius: 4px;
  font-family: 'Poppins',sans-serif;
  padding: 3em;
  text-align: center;
}

.c7 {
  display: none;
  width: 0;
}

.c2 {
  position: absolute;
  right: 0;
  top: 0;
}

.c3 {
  background: none;
  border: 0;
  cursor: pointer;
}

.c1 {
  margin: 1em;
  position: relative;
}

.c6 {
  color: #404040;
}

.c5 {
  max-height: 200px;
  max-width: 200px;
}

.c4 {
  display: inline-block;
}

<div
    class="c0"
    role="application"
  >
    <div
      class="c1"
    >
      <span
        class="c2"
      >
        <button
          aria-label="Discard"
          class="c3"
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 6L6 18"
              stroke="#151515"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
            <path
              d="M6 6L18 18"
              stroke="#151515"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
      </span>
      <div
        class="c4"
        data-testid="blur"
      >
        <img
          alt="Preview of uploaded image"
          class="c5"
          src="[object Object]"
        />
      </div>
      <div
        class="c6"
      >
        HangryCat
      </div>
    </div>
    <input
      accept="image/png,image/jpeg,text/plain,model/gltf-binary,.glb,.gltf"
      aria-invalid="false"
      aria-label="File selector"
      class="c7"
      type="file"
    />
  </div>
</DocumentFragment>
`;
